compliance/index/v4/node.go

```diff
@@ -0,0 +1,185 @@
+package v4
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/quay/claircore"
+	ccindexer "github.com/quay/claircore/indexer"
+	"github.com/quay/claircore/indexer/controller"
+	"github.com/quay/claircore/rhel"
+	rpm2 "github.com/quay/claircore/rpm"
+	"github.com/stackrox/rox/compliance/collection/compliance"
+	"github.com/stackrox/rox/compliance/collection/intervals"
+	v4 "github.com/stackrox/rox/generated/internalapi/scanner/v4"
+	"github.com/stackrox/rox/pkg/env"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/scannerv4/mappers"
+)
+
+var (
+	// The layer carries a hardcoded digest, as it is exclusively used for passing
+	// ClairCore checks, not for Scanners matching
+	layerDigest = fmt.Sprintf("sha256:%s", strings.Repeat("a", 64))
+	log         = logging.LoggerForModule()
+)
+
+type localNodeIndexer struct {
+}
+
+// NewNodeIndexer creates a new node indexer
+func NewNodeIndexer() compliance.NodeIndexer {
+	return &localNodeIndexer{}
+}
+
+// GetIntervals
+func (l *localNodeIndexer) GetIntervals() *intervals.NodeScanIntervals {
+	i := intervals.NewNodeScanIntervalFromEnv()
+	return &i
+}
+
+// IndexNode indexes a live fs.FS at the container mountpoint given in the basePath.
+func (l *localNodeIndexer) IndexNode(ctx context.Context) (r *v4.IndexReport, err error) {
+	report := &claircore.IndexReport{
+		Packages:      map[string]*claircore.Package{},
+		Environments:  map[string][]*claircore.Environment{},
+		Distributions: map[string]*claircore.Distribution{},
+		Repositories:  map[string]*claircore.Repository{},
+		Files:         map[string]claircore.File{},
+	}
+
+	layer, err := constructLayer(ctx, layerDigest, env.NodeScanningV4HostPath.Setting())
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		if tmpErr := layer.Close(); tmpErr != nil {
+			err = tmpErr
```

Comment:
This overwrites the pre-existing error. I'm thinking we should do `errors.Join(err, closeErr)` (and renaming `tmpErr` to `closeErr`)